JoonasHeinonen/escape-from-zordoom-prison
# Feature/aiming system

### Link to the ticket: [Aiming system](https://trello.com/c/ufPCISkE/37-aiming-system)

https://user-images.githubusercontent.com/33086486/192164657-48ecae30-8182-4382-bfd5-5aaa5b1e6bf8.mp4

**Changes:**
- Changed the aiming system from manual to automatic mouse targeting.
- Added shoot function for mouse and changed arrows for WASD buttons for the movement.

**How to test:**
 - Verify that the hand is rotating perfectly. (_Currently, unable to position the arm correctly as it affects the aiming system fatally._)
 - In terms of user experience, confirm that this form of gameplay and settings is optimal for the user experience.

**Documentation & sources:**
 - [Godot official documentation - Input examples, InputMap](https://docs.godotengine.org/en/stable/tutorials/inputs/input_examples.html#inputmap)
 - [Ray-casting from the screen : Look at the mouse in 3D [GODOT]
](https://www.youtube.com/watch?v=mmvIkkKJVlQ&list=PLJMAArawqfKqyl4ZyNQPv1U_3e0zNfY1C&index=7&t=113s&ab_channel=Nolkaloid)
